src/pykeen/datasets/base.py

```diff
@@ -379,17 +379,14 @@ def _help_cache(self, cache_root: Union[None, str, pathlib.Path]) -> pathlib.Pat
             :class:`pykeen.datasets.base.Dataset`.
         :returns: A path object for the calculated cache root directory
         """
-        if cache_root is None:
-            cache_root = PYKEEN_DATASETS
-        cache_root = pathlib.Path(cache_root).resolve()
-        cache_root = self._extend_cache_root(cache_root=cache_root)
-        cache_root.mkdir(parents=True, exist_ok=True)
+        cache_root = normalize_path(cache_root, *self._cache_sub_directories(), mkdir=True, default=PYKEEN_DATASETS)
         logger.debug("using cache root at %s", cache_root.as_uri())
         return cache_root
 
-    def _extend_cache_root(self, cache_root: pathlib.Path) -> pathlib.Path:
-        """Get appropriate cache sub-directory."""
-        return cache_root.joinpath(self.__class__.__name__.lower())
+    def _cache_sub_directories(self) -> Iterable[str]:
+        """Iterate over appropriate cache sub-directory."""
+        # TODO: use class-resolver normalize?
+        return self.__class__.__name__.lower()
```

Comment:
this looks like a bug, this returns the string instead of yields it. why does this need to be a function? it's only used once

Reply:
it is overridden by wk3l to share the cache between cn3l, wk3l15k and wk3l120k